src/multi-jvm/scala/akka/contrib/datareplication/ReplicatorSpec.scala

```diff
@@ -481,5 +480,32 @@ class ReplicatorSpec extends MultiNodeSpec(ReplicatorSpec) with STMultiNodeSpec
     enterBarrier("after-8")
   }
 
+  "check that a gossip update and a local update both cause a change event to emit with the merged data" in {
+    val changedProbe = TestProbe()
+
+    runOn(second) {
+      replicator ! Subscribe("H", changedProbe.ref)
+      replicator ! Update("H", ORMap.empty[Flag], writeTwo)(_ + ("a" -> Flag(enabled = false)))
+      changedProbe.expectMsgPF() { case Changed("H", h: ORMap[Flag] @unchecked) ⇒ h.entries } should be(Map("a" -> Flag(enabled = false)))
+    }
+
+    enterBarrier("update-h1")
+
+    runOn(first) {
+      replicator ! Update("H", ORMap.empty[Flag], writeTwo)(_ + ("a" -> Flag(enabled = true)))
+    }
+
+    runOn(second) {
+      replicator ! Update("H", ORMap.empty[Flag], writeTwo)(_ + ("b" -> Flag(enabled = true)))
+      changedProbe.expectMsgPF() {
+        case Changed("H", ORMap(initialEntries)) if initialEntries.size == 1 ⇒
+          changedProbe.expectMsgPF() { case Changed("H", ORMap(entries)) ⇒ entries }
+        case Changed("H", ORMap(entries)) ⇒
+          entries
```

Comment:
This test sometimes fails, here it can be Map(a -> Flag(false), b -> Flag(true)), i.e. the local update of "b" is published but the remote update "a" -> true has not arrived yet. I'm rewriting the test to be more deterministic.